py/server/deephaven/experimental/s3.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+"""
+    This module is useful for reading and writing files stored in S3. Importing this module requires the S3 specific
+    extensions to be included in the class path.
+"""
+import datetime
+from typing import Optional, Union
+
+import jpy
+import numpy as np
+import pandas as pd
+
+from deephaven import time
+from deephaven._wrapper import JObjectWrapper
+from deephaven.dtypes import Duration
+
+
+class S3Instructions(JObjectWrapper):
```

Comment:
Most of the instructions are common for both read and write, like region_name, timeouts, etc. So I wanted to keep a generic name which can be extended to include writing specific instructions as we add support for writing to S3.
Same is the philosophy behind `ParquetInstructions` class.

Reply:
ok.  Should there be a `class S3ReadInstructions(S3Instructions)` created now, or will `S3ReadInstructions` always be identical to `S3Instructions`?  Just asking for future API stability reasons.

Since this is still experimental, I won't push the question hard, but think about it.

Reply:
At this point, our plan is to focus on reads first, so S3ReadInstructions will be similar to S3Instructions for some time, and then later when we add writes, we may add some write specific instructions.
For now, I wanted to keep the structure similar to the Java code. There also, we have just a single S3Instructions.